project-extender/src/providers/jira-server-provider/JiraServerProvider.ts

```diff
@@ -26,29 +42,88 @@ class JiraServerProvider implements ProviderApi {
   }
 
   async isLoggedIn(): Promise<void> {
+    const response = fetch(
+      `http://${this.requestBody.url}/rest/auth/1/session`,
+      {
+        method: "GET",
+        headers: {
+          Accept: "application/json",
+          Authorization: `Basic ${Buffer.from(
+            `${this.requestBody.username}:${this.requestBody.password}`
+          ).toString("base64")}`,
+        },
+      }
+    )
+    const GoodResponse = await response
     return new Promise((resolve, reject) => {
-      this.provider
-        ?.getCurrentUser()
-        .then(() => resolve())
-        .catch(() => reject())
+      if (GoodResponse.status === 200) resolve()
+      if (GoodResponse.status === 401) {
+        reject(new Error("Wrong Username or Password"))
+      }
+      response.catch((err) => {
+        if (err.message === "Couldn't connect to server")
+          reject(new Error("Wrong URL"))
+      })
```

Comment:
look at the refactored `logout()` method and refactor this one too 👍

Reply:
fixed

Reply:
This was not fixed